core/schema_utils.py

```diff
@@ -682,3 +682,27 @@ def is_valid_username_string(obj: str) -> bool:
             return True
         except utils.ValidationError:
             return False
+
+    @staticmethod
+    def should_end_with(
+        obj: str,
+        choices: List[str],
+        match_case: Optional[bool] = False
+    ) -> bool:
+        """Checks if the given obj (a string) ends with any one
+        of the given choices.
+
+        Args:
+            obj: str. The string to verify.
+            choices: List[str]. All valid choices.
+            match_case: bool. Whether the match is case sensitive.
+
+        Returns:
+            bool. Whether the given object is a filename with valid
+            audio extension.
+        """
+
+        if match_case is False:
+            obj.lower()
```

Comment:
This won't change the `obj` value, right? Add a proper test case for the match_case condition!

Reply:
Yes you are right. I need to assign it back to `obj` to actually change the value. I already covered test cases for match_case conditions in the test file.

Reply:
>  I already covered test cases for match_case conditions in the test file.

So why it wasn't failing earlier in case the old changes were incorrect?

Reply:
It wasn't failing earlier because of inadequate test cases, specifically for `match_case=False` as you pointed out. I will add new tests to cover this.

Reply:
> why it wasn't failing earlier in case the old changes were incorrect?

It wasn't failing earlier because of inadequate test cases, specifically for `match_case=False` as you pointed out. I will add new tests to cover this.

Reply:
> Make sure to have a test for `match_case=False`

Done.